sympy/integrals/laplace.py

```diff
@@ -589,8 +589,21 @@ def _laplace_rule_delta(f, t, s):
             debug('      rule: multiply with DiracDelta')
             loc = ma2[a]/ma2[b]
             if re(loc) >= 0 and im(loc) == 0:
-                r = exp(-ma2[a]/ma2[b]*s)*ma1[z].subs(t, ma2[a]/ma2[b])/ma2[b]
-                return (r, S.NegativeInfinity, S.true)
+                fn = exp(-ma2[a]/ma2[b]*s)*ma1[z]
+                print('--', fn)
+                r = fn.subs(t, ma2[a]/ma2[b])/ma2[b]
+                print('--', r)
+                if r == nan and (fn.has(sin, cos)):
+                    # Then it may be possible that a sinc() is present in the
```

Comment:
In general I think it is better to try to avoid doing anything that might give a nan ahead of time rather than checkinf for nan retrospectively. Maybe it is better to use something like `as_numer_denom` and then handle the numerator and denominator separately?

Essentially what you are doing here is computing a simple limit but I don't think that we would want to invoke the limit code here.

Perhaps there should be a pattern somewhere that can do that sinc rewrite early on.

Reply:
Very good idea, thanks, I have re-written it in a better way.